daemon/src/identify/dialer.rs

```diff
@@ -0,0 +1,192 @@
+            }
+        };
+
+        let wire_version = identity_info.wire_version.clone();
+        let daemon_version = identity_info.daemon_version.clone();
+        let environment = identity_info.environment;
+
+        NUM_LIBP2P_CONNECTIONS_GAUGE
+            .with(&HashMap::from([
+                (WIRE_VERSION_LABEL, wire_version.as_str()),
+                (DAEMON_VERSION_LABEL, daemon_version.as_str()),
+                (ENVIRONMENT_LABEL, environment.to_string().as_str()),
+            ]))
+            .inc();
+
+        tracing::info!(%peer_id, %daemon_version, %environment, %wire_version, "New identify message received");
+        self.identities.insert(peer_id, identity_info);
+    }
+}
+
+#[xtra_productivity(message_impl = false)]
+impl Actor {
+    async fn handle_connections_established(
+        &mut self,
+        msg: endpoint::ConnectionEstablished,
+        ctx: &mut Context<Self>,
+    ) {
+        let peer_id = msg.peer;
+        let endpoint = self.endpoint.clone();
+        let this = ctx.address().expect("we are alive");
+
+        let request_peer_identity_fut = async move {
+            let stream = endpoint
+                .send(OpenSubstream::single_protocol(peer_id, PROTOCOL))
+                .await??;
+
+            let peer_identity = protocol::recv(stream).await?;
+
+            this.send(PeerIdentityReceived {
+                peer_id,
+                peer_identity,
+            })
+            .await?;
+
+            anyhow::Ok(())
+        };
+
+        let err_handler = move |e| async move {
+            tracing::debug!(%peer_id, "Identity protocol failed upon request: {e:#}")
+        };
+
+        if let Err(e) = self
+            .spawner
+            .as_ref()
+            .expect("some after started")
+            .send_async_safe(SpawnFallible::new(request_peer_identity_fut, err_handler))
+            .await
+        {
+            tracing::error!("Failed to spawn identity task: {e:#}");
+        };
+    }
+
+    async fn handle_connections_dropped(&mut self, msg: endpoint::ConnectionDropped) {
+        let peer_id = msg.peer;
+        tracing::trace!(%peer_id, "Remove peer-info because connection dropped");
+        if let Some(identity_info) = self.identities.remove(&peer_id) {
+            NUM_LIBP2P_CONNECTIONS_GAUGE
+                .with(&HashMap::from([
+                    (WIRE_VERSION_LABEL, identity_info.wire_version.as_str()),
+                    (DAEMON_VERSION_LABEL, identity_info.daemon_version.as_str()),
+                    (
+                        ENVIRONMENT_LABEL,
+                        identity_info.environment.to_string().as_str(),
+                    ),
+                ]))
+                .dec();
+        }
+    }
+}
+
+const WIRE_VERSION_LABEL: &str = "wire_version";
+const DAEMON_VERSION_LABEL: &str = "daemon_version";
+const ENVIRONMENT_LABEL: &str = "environment";
+
+static NUM_LIBP2P_CONNECTIONS_GAUGE: conquer_once::Lazy<prometheus::IntGaugeVec> =
```

Comment:
Note: we could try to re-use the old gauge (i.e. `NUM_P2P_CONNECTIONS_GAUGE`) in the following way:

1. The maker has to still collect information for old and new connections but uses the same gauge.
2. We remove the old connection setup from the taker as part of this PR.

Reply:
I think this is a good idea. otherwise, it gets unnecessarily complicated on grafana side.

Reply:
I am not convonced of this anymore, unless we manage to finish the `order` protocol first and only then change this PR. Otherwise there are too many dependencies to the old connection to be able to remove this cleanly. 
Generally the time invest to re-use the old gauge might not be worth it, might be simpler to adapt the metric in Grafana to the new one and then just remove the complete old gauge.